### PR TITLE
adding my keymap

### PR DESCRIPTION
**Pull request by MVi06**
## Description

It’s the keymap I use. Based on the bépo french layout.

### CODE REVIEW

File: keyboards/crkbd/keymaps/mvi/config.h [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+#ifdef RGBLIGHT_ENABLE
+    #undef RGBLED_NUM
+    #define RGBLED_NUM 27
+    #define RGBLIGHT_ANIMATIONS
+    #define RGBLIGHT_SLEEP
+    #define RGBLIGHT_SPLIT
+    #define RGBLIGHT_LIMIT_VAL 120
+    #define RGBLIGHT_HUE_STEP 10
+    #define RGBLIGHT_SAT_STEP 17
+    #define RGBLIGHT_VAL_STEP 17
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/config.h:69]: Nit pick:
```suggestion
#    undef RGBLED_NUM
#    define RGBLED_NUM 27
#    define RGBLIGHT_ANIMATIONS
#    define RGBLIGHT_SLEEP
#    define RGBLIGHT_SPLIT
#    define RGBLIGHT_LIMIT_VAL 120
#    define RGBLIGHT_HUE_STEP 10
#    define RGBLIGHT_SAT_STEP 17
#    define RGBLIGHT_VAL_STEP 17
```

File: keyboards/crkbd/keymaps/mvi/config.h [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+#endif
+
+#define OLED_FONT_H "keyboards/crkbd/keymaps/soundmonster/glcdfont.c"
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/config.h:72]:
```suggestion
#define OLED_FONT_H "keyboards/crkbd/keymaps/mvi/glcdfont.c"
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 3]
```diff
@@ -0,0 +1,909 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include "action_layer.h"
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:21]:
```suggestion
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 4]
```diff
@@ -0,0 +1,909 @@
...
+#include "keymap_bepo.h"
+
+#ifndef MOUSEKEY_H
+#include "mousekey.h"
+#endif
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:26]: Is this needed?

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 5]
```diff
@@ -0,0 +1,909 @@
...
+// super alt/tab macro
+bool is_alt_tab_active = false; // ADD this near the begining of keymap.c
+uint16_t alt_tab_timer = 0;     // we will be using them soon.extern keymap_config_t keymap_config;
+
+uint16_t mu_cced_timer = 0;
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:37]:
```suggestion
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 6]
```diff
@@ -0,0 +1,909 @@
...
+  _BEPO,
+  _NUMPAD,
+  _LOWER,
+  _RAISE,
+  _ADJUST,
+  _TOUCHCURSOR,
+  _MATHS
+};
+
+int current_base_layer = _BEPO;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:59]:
```suggestion
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 7]
```diff
@@ -0,0 +1,909 @@
...
+     KC_ESC,  _______, _______, _______, _______, _______,                      _______, BEPO,    NUMPAD,  _______, _______, DEBUG,
+  //|--------+--------+--------+--------+--------+--------|                    |--------+--------+--------+--------+--------+--------|
+     _______, _______, _______, _______, _______, _______,                      _______, _______, _______, _______, _______, _______,
+  //|--------+--------+--------+--------+--------+--------+--------|  |--------+--------+--------+--------+--------+--------+--------|
+                                         _______, _______, _______,    _______, _______, _______
+                                      //`--------------------------'  `--------------------------'
+  )
+};
+
+int RGB_current_mode;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:230]:
```suggestion
uint8_t RGB_current_mode;
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 8]
```diff
@@ -0,0 +1,909 @@
...
+
+// Setting ADJUST layer RGB back to default
+void update_tri_layer_RGB(uint8_t layer1, uint8_t layer2, uint8_t layer3) {
+  if (IS_LAYER_ON(layer1) && IS_LAYER_ON(layer2)) {
+    layer_on(layer3);
+  } else {
+    layer_off(layer3);
+  }
+} 
+
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:245]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

Same with the tri layer function 
```suggestion
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 9]
```diff
@@ -0,0 +1,909 @@
...
+void matrix_init_user(void) {
+    #ifdef RGBLIGHT_ENABLE
+      RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:248]:
```suggestion
      RGB_current_mode = rgblight_get_mode();
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 10]
```diff
@@ -0,0 +1,909 @@
...
+    if(layer_state_is(_ADJUST)) {
+        oled_write_P(adjust_layer, false);
+    } else if(layer_state_is(_LOWER)) {
+        oled_write_P(lower_layer, false);
+    } else if(layer_state_is(_RAISE)) {
+        oled_write_P(raise_layer, false);
+    } else {
+        oled_write_P(default_layer, false);
+    }
+    switch(current_base_layer) {
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:424]:
```suggestion
    switch(get_highest_layer(default_layer_state)) {
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 11]
```diff
@@ -0,0 +1,909 @@
...
+#ifdef OLED_DRIVER_ENABLE
+        oled_timer = timer_read32();
+#endif
+    // set_timelog();
+  }
+
+  switch (keycode) {
+    case BEPO:
+      if (record->event.pressed) {
+	current_base_layer = _BEPO;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:487]:
```suggestion
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 12]
```diff
@@ -0,0 +1,909 @@
...
+        persistent_default_layer_set(1UL<<_BEPO);
+      }
+      return false;
+    case NUMPAD:
+      current_base_layer = _NUMPAD;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:492]:
```suggestion
```

File: keyboards/crkbd/keymaps/mvi/keymap.c [hunk 13]
```diff
@@ -0,0 +1,909 @@
...
+        oled_timer = timer_read32();
+#endif
+    // set_timelog();
+  }
+
+  switch (keycode) {
+    case BEPO:
+      if (record->event.pressed) {
+	current_base_layer = _BEPO;
+        persistent_default_layer_set(1UL<<_BEPO);
```
Comment (drashna) [inline keyboards/crkbd/keymaps/mvi/keymap.c:488]:
```suggestion
        set_single_persistent_default_layer(_BEPO);
```

File: keyboards/planck/keymaps/a2b/rules.mk [hunk 14]
```diff
@@ -0,0 +1,5 @@
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/planck/keymaps/a2b/rules.mk:1]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/planck/keymaps/mvi/config.h [hunk 15]
```diff
@@ -0,0 +1,52 @@
+#pragma once
```
Comment (drashna) [inline keyboards/planck/keymaps/mvi/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/planck/keymaps/mvi/keymap.c [hunk 16]
```diff
@@ -0,0 +1,817 @@
...
+ */
+
+#include QMK_KEYBOARD_H
+#include "muse.h"
+#include "keymap_bepo.h"
+#include "sendstring_bepo.h"
+
+#ifndef MOUSEKEY_H
+#include "mousekey.h"
+#endif
```
Comment (drashna) [inline keyboards/planck/keymaps/mvi/keymap.c:27]: Is this needed?

File: keyboards/planck/keymaps/mvi/keymap.c [hunk 17]
```diff
@@ -0,0 +1,817 @@
...
+
+// super alt/tab macro
+bool is_alt_tab_active = false; // ADD this near the begining of keymap.c
+uint16_t alt_tab_timer = 0;     // we will be using them soon.
+
+uint16_t mu_cced_timer = 0;
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/planck/keymaps/mvi/keymap.c:35]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
It looks like you need to rebase your branch here.

File: keyboards/crkbd/keymaps/mvi/config.h [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+#    undef RGBLED_NUM
+#    define RGBLED_NUM 27
+#    define RGBLIGHT_ANIMATIONS
+#    define RGBLIGHT_SLEEP
+#    define RGBLIGHT_SPLIT
+#    define RGBLIGHT_LIMIT_VAL 120
+#    define RGBLIGHT_HUE_STEP 10
+#    define RGBLIGHT_SAT_STEP 17
+#    define RGBLIGHT_VAL_STEP 17
+<<<<<<< HEAD
```
Comment (tzarc) [inline keyboards/crkbd/keymaps/mvi/config.h:72]: I think you might've had a problematic merge. Might want to go through your entire PR again.

**Review by tzarc — CHANGES_REQUESTED** [review]

**Comment by MVi06** [comment]
I will try to do a cleaner pull request. Thanks for the advices.

**Pull request closed without merging**